src/main/java/datawave/data/type/util/NumericalEncoder.java

```diff
@@ -115,133 +128,36 @@ public static BigDecimal decode(String input) {
-        positiveExponents[9] = "+J";
-        positiveExponents[10] = "+K";
-        positiveExponents[11] = "+L";
-        positiveExponents[12] = "+M";
-        positiveExponents[13] = "+N";
-        positiveExponents[14] = "+O";
-        positiveExponents[15] = "+P";
-        positiveExponents[16] = "+Q";
-        positiveExponents[17] = "+R";
-        positiveExponents[18] = "+S";
-        positiveExponents[19] = "+T";
-        positiveExponents[20] = "+U";
-        positiveExponents[21] = "+V";
-        positiveExponents[22] = "+W";
-        positiveExponents[23] = "+X";
-        positiveExponents[24] = "+Y";
-        positiveExponents[25] = "+Z";
-        positiveExponents[26] = "+a";
-        positiveExponents[27] = "+b";
-        positiveExponents[28] = "+c";
-        positiveExponents[29] = "+d";
-        positiveExponents[30] = "+e";
-        positiveExponents[31] = "+f";
-        positiveExponents[32] = "+g";
-        positiveExponents[33] = "+h";
-        positiveExponents[34] = "+i";
-        positiveExponents[35] = "+j";
-        positiveExponents[36] = "+k";
-        positiveExponents[37] = "+l";
-        positiveExponents[38] = "+m";
-        positiveExponents[39] = "+n";
-        positiveExponents[40] = "+o";
-        positiveExponents[41] = "+p";
-        positiveExponents[42] = "+q";
-        positiveExponents[43] = "+r";
-        positiveExponents[44] = "+s";
-        positiveExponents[45] = "+t";
-        positiveExponents[46] = "+u";
-        positiveExponents[47] = "+v";
-        positiveExponents[48] = "+w";
-        positiveExponents[49] = "+x";
-        positiveExponents[50] = "+y";
-        positiveExponents[51] = "+z";
-        
-        positiveNumsEncodeToIntExponentsMap = new HashMap<String,String>();
-        positiveNumsIntToEncodeExponentsMap = new HashMap<String,String>();
-        for (int j = 0; j < positiveExponents.length; j++) {
+    private static Map<String,String> createExponentMap(String[] exponents) {
+        Map<String,String> map = new HashMap<>();
+        for (int j = 0; j < exponents.length; j++) {
             int exponent = j - 26;
-            positiveNumsEncodeToIntExponentsMap.put(positiveExponents[j], String.valueOf(exponent));
-            positiveNumsIntToEncodeExponentsMap.put(String.valueOf(exponent), positiveExponents[j]);
+            map.put(exponents[j], String.valueOf(exponent));
         }
+        return map;
+    }
+    
+    private static Map<String,String> invertMap(Map<String,String> map) {
+        return map.entrySet().stream().collect(Collectors.toMap(Map.Entry::getValue, Map.Entry::getKey));
+    }
+    
+    private static void initPositiveExponents() {
+        // The order of the encoded characters here maps directly to how their corresponding exponent value is calculated, and must not be changed.
+        String[] exponents = new String[] {"+A", "+B", "+C", "+D", "+E", "+F", "+G", "+H", "+I", "+J", "+K", "+L", "+M", "+N", "+O", "+P", "+Q", "+R", "+S",
```

Comment:
Since you went through the trouble of simplifying the generation of these exponents, I wound if we can generate then without having a hardcoded list (which could contain a typo that we simply miss).  We can programmatically cycle through the characters A to Z and then a to z.

Reply:
What was your conclusion on this one?

Reply:
I am still planning on doing it, I just got pulled into other things. I will try to get this done today.

Reply:
This is complete.